### PR TITLE
melpaPackages.telega: Fix build

### PR DESCRIPTION
**Pull request by SeTSeR**
###### Motivation for this change
From version 0.7.9 telega.el uses `pkg-config` to configure compiler and linker flags for server part. This change ensures that `pkg-config` is available at build time and fixes the build.

###### Things done

pkg-config is added to `nativeBuildInputs` of `telega` package

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/emacs-modes/melpa-packages.nix [hunk 1]
```diff
@@ -341,6 +341,7 @@ let
         # Telega has a server portion for it's network protocol
         telega = super.telega.overrideAttrs (old: {
           buildInputs = old.buildInputs ++ [ pkgs.tdlib ];
+          nativeBuildInputs = [ external.pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/emacs-modes/melpa-packages.nix:344]:
```suggestion
          nativeBuildInputs = [ external.pkg-config ];
```

Comment (SeTSeR) [reply pkgs/applications/editors/emacs-modes/melpa-packages.nix:344]: As far as I can see, both variants are valid. Other MELPA packages with native parts, for example pdf-tools, also use `pkgconfig`. Is there any difference between these packages?

Comment (SuperSandro2000) [reply pkgs/applications/editors/emacs-modes/melpa-packages.nix:344]: pkgconfig is an alias to pkg-config which we want to remove in the future and currently we are doing treewide PRs to remove the old alias. See eg #109595. Other packages should be moved away from in the near future and any new addition should not use the old alias.

Comment (SeTSeR) [reply pkgs/applications/editors/emacs-modes/melpa-packages.nix]: Thank you for explanation, fixed

**Pull request merged**